controllers/controlplane.go

```diff
@@ -62,6 +63,10 @@ func (c *ControlPlane) MachineWithDeleteAnnotation(machines collections.Machines
 
 // MachinesNeedingRollout return a list of machines that need to be rolled out.
 func (c *ControlPlane) MachinesNeedingRollout() collections.Machines {
+	if c.TCP.Spec.RolloutStrategy == nil && c.TCP.Spec.RolloutStrategy.Type == v1alpha3.OnDeleteStrategyType {
```

Comment:
:facepalm: should be

```suggestion
	if c.TCP.Spec.RolloutStrategy != nil && c.TCP.Spec.RolloutStrategy.Type == v1alpha3.OnDeleteStrategyType {
```